registration-processor/registration-processor-common-camel-bridge/src/test/java/io/mosip/registration/processor/camel/bridge/PauseFlowPredicateTest.java

```diff
@@ -1,185 +1,184 @@
+import io.mosip.kernel.core.util.DateUtils;
+import io.mosip.registration.processor.camel.bridge.intercepter.PauseFlowPredicate;
+import io.mosip.registration.processor.camel.bridge.model.Setting;
+import io.mosip.registration.processor.core.abstractverticle.MessageDTO;
+import io.mosip.registration.processor.core.constant.LoggerFileConstant;
+import io.mosip.registration.processor.core.logger.RegProcessorLogger;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(VertxUnitRunner.class)
+@PowerMockIgnore({ "javax.management.*", "javax.net.ssl.*", "com.sun.org.apache.xerces.*", "javax.xml.*", "org.xml.*" })
+
+@AutoConfigureMockMvc
+public class PauseFlowPredicateTest {
+
+	private PauseFlowPredicate pauseFlowPredicate;
+
+	private ObjectMapper objectMapper;
+
+	private Exchange exchange;
+
+	private static final Logger LOGGER = RegProcessorLogger.getLogger(PauseFlowPredicateTest.class);
+
+	@Before
+	public void setup() {
+		objectMapper = new ObjectMapper();
+		pauseFlowPredicate = new PauseFlowPredicate();
+
+		Setting[] settings = null;
+		try {
+			settings = objectMapper.readValue(
+					"[{\"matchExpression\": \"$.tags[?(@['META_INFO-META_DATA-machineId'] == '10074')]\",\"pauseFor\": 432000,\"defaultResumeAction\": \"ResumeFromBeginning\",\"fromAddress\": \"bio-debup-bus-out\",\"resumeRemoveTags\": \"HOTLISTED,BIOMETRICS_EXCEPTION\"},{\"matchExpression\": \"$.tags[?(@['META_INFO-META_DATA-machineId'] == '10074')]\",\"pauseFor\": 432000,\"defaultResumeAction\": \"StopProcessing\",\"fromAddress\": \"bio-debup-bus-out\",\"resumeRemoveTags\": \"HOTLISTED,BIOMETRICS_EXCEPTION\"}]",
+					Setting[].class);
+		} catch (IOException e) {
+			LOGGER.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(), "",
+					"RoutePredicate::exception " + e.getMessage());
+		}
+		ReflectionTestUtils.setField(pauseFlowPredicate, "settings", settings);
+		ReflectionTestUtils.setField(pauseFlowPredicate, "objectMapper", objectMapper);
+		
+		Endpoint endpoint = new DefaultEndpoint() {
+
+			@Override
+			public String toString() {
+
+				return "bio-debup-bus-out";
+			}
+
+			@Override
+			public boolean isSingleton() {
+				// TODO Auto-generated method stub
+				return false;
+			}
+
+			@Override
+			public Producer createProducer() throws Exception {
+				// TODO Auto-generated method stub
+				return null;
+			}
+
+			@Override
+			public Consumer createConsumer(Processor processor) throws Exception {
+				// TODO Auto-generated method stub
+				return null;
+			}
+		};
+
+		exchange = new DefaultExchange(endpoint);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Test
+	public void testRoutePredicatePositive() throws Exception {
+		MessageDTO messageDTO = new MessageDTO();
+		messageDTO.setRid("10002100741000120201231071308");
+		Map<String, String> tags = new HashMap<>();
+		tags.put("META_INFO-META_DATA-machineId", "10074");

```

Comment:
Lets use the same HOTLISTED tag for all the test as before and change the matchExpression accordingly

Reply:
Fixed test cases